src/Symfony/Component/Security/Core/User/AttributesBasedUserProviderInterface.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\User;
+
+use Symfony\Component\Security\Core\Exception\UserNotFoundException;
+
+/**
+ * Overrides UserProviderInterface to add a "attributes" argument on loadUserByIdentifier.
+ * This is particularly useful with self-contained access tokens.
+ *
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ */
+interface AttributesBasedUserProviderInterface extends UserProviderInterface
```

Comment:
This is the only piece of code I'm not entirely sure about but for which I don't have any alternative to propose either. I'm tempted to suggest marking it as `@experimental in 6.3` to allow for someone to come up with a different approach or just to tweak the implementation.

Reply:
@vincentchalamon Can you do it? Maybe we should make the whole feature experimental for now?